src/main/java/TextView.java

```diff
@@ -111,28 +111,42 @@ void drawBoard(Board board)
 	 * @return String representation of piece.
 	 * @throws IllegalArgumentException For invalid piece type.
 	 */
-	String getAbbreviation(Piece piece) throws IllegalArgumentException
+	String getAbbreviation(Piece piece)
 	{
 		PieceType pieceType = piece.getType();
+		String pieceAbr ="";
 		switch (pieceType)
 		{
 		case BISHOP:
-			return " BI ";
+			pieceAbr = " BI ";
+			break;
 		case KING:
-			return " KI ";
+			pieceAbr = " KI ";
+			break;
 		case KNIGHT:
-			return " KN ";
+			pieceAbr = " KN ";
+			break;
 		case PAWN:
-			return " PA ";
+			pieceAbr = " PA ";
+			break;
 		case QUEEN:
-			return " QU ";
+			pieceAbr = " QU ";
+			break;
 		case ROOK:
-			return " RO ";
+			pieceAbr = " RO ";
+			break;
 		case NO_PIECE:
-			return " .. ";
+			pieceAbr = " .. ";
+			break;
 		default:
-			throw new IllegalArgumentException("Invalid piece type.");
-
+			
+		}	
+		if (piece.getColor() == PieceColor.BLACK)
+		{
+			pieceAbr = pieceAbr.toLowerCase();
```

Comment:
- [ ] Minor thing. It's slightly cleaner to change this line to `return peiceAbr.toLowerCase()`, instead of reassigning it and returning it on the next line.